r/R/matrix.R

```diff
@@ -1109,15 +1109,15 @@ setMethod("rbind2", signature(x = "IterableMatrix", y = "IterableMatrix"), funct
   if (x@transpose != y@transpose) stop("Cannot merge matrices with different internal transpose states.\nPlease use transpose_storage_order().")
   if (matrix_type(x) != matrix_type(y)) {
     # upcast if mismatching types
-    type_order <- c("uint32_t", "float", "double")
-    new_type <- type_order[max(match(matrix_type(x), type_order), match(matrix_type(y), type_order))]
     rlang::warn(
-      sprintf("First matrix type: %s, second matrix type: %s.\nUpcasting to %s to match matrix types", 
-        matrix_type(x), matrix_type(y), new_type
+      sprintf(
+        paste0("Mismatching matrix types when calling %sbind().\n",
+        "First matrix type: %s, second matrix type: %s.\nUpcasting to double to match matrix types"), 
+        ifelse(x@transpose, "c", "r"), matrix_type(x), matrix_type(y)
       )
     )
-    x <- convert_matrix_type(x, new_type)
-    y <- convert_matrix_type(y, new_type)
+    if (matrix_type(x) != "double") x <- convert_matrix_type(x, "double")
```

Comment:
Minor nit: you technically don't need the if statement, as `convert_matrix_type` already no-ops if the matrix is already the desired type